recipes/mosquitto/2.x/conanfile.py

```diff
@@ -40,113 +42,90 @@ class Mosquitto(ConanFile):
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    def _patch_sources(self):
-        replace_in_file(self, os.path.join(self._source_subfolder, "client", "CMakeLists.txt"), "static)", "static ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "client", "CMakeLists.txt"), "quitto)", "quitto ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "static)", "static ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "quitto)", "quitto ${CONAN_LIBS})")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_passwd", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "apps", "mosquitto_ctrl", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"), "OPENSSL_LIBRARIES", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "src", "CMakeLists.txt"), "MOSQ_LIBS", "CONAN_LIBS")
-        replace_in_file(self, os.path.join(self._source_subfolder, "include", "mosquitto.h"), "__declspec(dllimport)", "")
-        replace_in_file(self, os.path.join(self._source_subfolder, "lib", "cpp", "mosquittopp.h"), "__declspec(dllimport)", "")
-        # dynlibs for apple mobile want code signatures and that will not work here
-        # this would actually be the right patch for static builds also, but this would have other side effects, so
-        if(self.settings.os in ["iOS", "watchOS", "tvOS"]):
-            replace_in_file(self, os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"), "SHARED", "")
-            replace_in_file(self, os.path.join(self._source_subfolder, "lib", "cpp", "CMakeLists.txt"), "SHARED", "")
+            tc.variables["WITH_CJSON"] = self.options.get_safe("cjson")
+        tc.variables["WITH_BROKER"] = self.options.broker
+        tc.variables["WITH_APPS"] = self.options.apps
+        tc.variables["WITH_PLUGINS"] = False
+        tc.variables["WITH_LIB_CPP"] = self.options.build_cpp
+        tc.variables["WITH_THREADING"] = not is_msvc(self) and self.options.threading
+        tc.variables["WITH_WEBSOCKETS"] = self.options.get_safe("websockets", False)
+        tc.variables["STATIC_WEBSOCKETS"] = self.options.get_safe("websockets", False) and not self.dependencies["libwebsockets"].options.shared
+        tc.variables["DOCUMENTATION"] = False
+        tc.variables["CMAKE_INSTALL_SYSCONFDIR"] = os.path.join(self.package_folder, "res").replace("\\", "/")
+        tc.variables["CMAKE_EXPORT_ALL_SYMBOLS"] = True
```

Comment:
```suggestion
        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```

@franramirez688 my bad, the correct directive is [CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS](https://cmake.org/cmake/help/v3.4/variable/CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.html)